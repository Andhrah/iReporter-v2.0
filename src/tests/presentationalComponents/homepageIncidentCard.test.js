import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import HomepageIncidentCard from '../../components/presentational/homepageIncidentCard';

const props = {
  incident: {
    corruption_methods: ['theft'],
    comment: 'this is ared flag',
  },
  type: 'intervention',
};

describe('homepageIncidentCard test suite', () => {
  it('renders correctly', () => {
    const wrapper = mount(
      <MemoryRouter>
        <HomepageIncidentCard {...props} />
      </MemoryRouter>
    );
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});
