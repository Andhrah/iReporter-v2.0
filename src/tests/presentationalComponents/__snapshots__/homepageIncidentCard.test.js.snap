// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homepageIncidentCard test suite renders correctly 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "aitxge",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "aitxge",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <HomepageIncidentCard
      incident={
        Object {
          "comment": "this is ared flag",
          "corruption_methods": Array [
            "theft",
          ],
        }
      }
      type="intervention"
    >
      <div
        className="incident-container-all list homepage"
      >
        <Image
          alt="red-flag incident"
          src="http://lorempixel.com/400/400"
        >
          <img
            alt="red-flag incident"
            className="red-flag-image"
            src="http://lorempixel.com/400/400"
          />
        </Image>
        <div
          className="home-comment"
        >
          <h4 />
          <p>
            this is ared flag
          </p>
        </div>
        <div>
          <Link
            className="read-more"
            to="redflag"
          >
            <a
              className="read-more"
              href="/redflag"
              onClick={[Function]}
            >
              Read more
               
              <i
                className="fas fa-angle-right"
              />
            </a>
          </Link>
        </div>
      </div>
    </HomepageIncidentCard>
  </Router>
</MemoryRouter>
`;
